### DISCUSSION TITLE
app.upload_trailer no longer working

### DISCUSSION THREAD

**Discussion by firegate666** [fastlane/fastlane#17816] [General] [answered]
Before I was using the old Tunes api to upload trailers to the app store. I am no setting up a new project and wanted to re-use my old scripts, but it seems apis have changed. So I was reading about the new ConnectAPI and figured out from the docs the new commands to login, find an app and get an edit version. But I can't find the replacement for uploading trailers?

    app = Spaceship::ConnectAPI::App.find(bundle_id)
    app_version = app.get_edit_app_store_version(platform:Spaceship::ConnectAPI::Platform::IOS)
    
    #app_version.upload_trailer(path_to_trailer, 1, "en-US", "iphone65", "00.00")
    #app_version.save!

**Comment by rogerluan** [comment]
Hey @firegate666 👋 

I'm afraid we haven't implemented AppTrailer upload for the new ConnectAPI yet, so you'll have to use the old API for now.

Hope this works out for you!

**Reply by firegate666** [reply]
Really? I invested some time yesterday and found everything I needed. I put together an example and pasted it on gist

https://gist.github.com/firegate666/9c6a0139a2efe6996a2ef8c1b31e7b94

**Reply by rogerluan** [reply]
Apologies, @firegate666 ! I didn't search for `app_preview` 😅 Glad you found it - and thanks for sharing that gist! ❤️

**Comment by firegate666** [comment, marked as answer]
The new process is slightly more complicated. Instead of calling the upload trailer method, you will have to get the app, get its localizations, there get the app preview sets and then you can upload everything.

I pasted a link with my working example on git and linked it above.
